packages/driver/test/cypress/integration/commands/navigation_spec.coffee

```diff
@@ -567,6 +567,15 @@ describe "src/cy/commands/navigation", ->
               "http://localhost:3500/fixtures/dimensions.html"
             ])
 
+    ## https://github.com/cypress-io/cypress/issues/1311
+    it "doesn't invoke onLoad callback when visiting the same URL with hashes", ->
+      cy
+        .visit("http://localhost:3500/fixtures/generic.html#foo")
+        .visit("http://localhost:3500/fixtures/generic.html#foo", {
+          onLoad: () ->
+            throw new Error("should not have called onLoad")
+        })
```

Comment:
also worth testing that the yielded subject is the `window` object by adding a `.then((win) => {})` and asserting on something that gives you assurance / guarantees the window did not reload or change.